tests/semanticdb/metac.expect

```diff
@@ -2016,6 +2016,60 @@ Occurrences:
 Synthetics:
 [5:4..5:8):List => *.apply[Int]
 
+expect/MatchType.scala
+----------------------
+
+Summary:
+Schema => SemanticDB v4
+Uri => MatchType.scala
+Text => empty
+Language => Scala
+Symbols => 10 entries
+Occurrences => 29 entries
+
+Symbols:
+example/MatchType$package. => final package object example extends Object { self: example.type => +3 decls }
+example/MatchType$package.Concat# => type Concat [typeparam Xs  <: Tuple, covariant typeparam Ys  <: Tuple] = Xs match { EmptyTuple => Ys, *:[x, xs] => *:[x, Concat[xs, Ys]] }
+example/MatchType$package.Concat#[Xs] => typeparam Xs  <: Tuple
+example/MatchType$package.Concat#[Ys] => covariant typeparam Ys  <: Tuple
+example/MatchType$package.Elem# => type Elem [typeparam X ] = X match { String => Char, Array[t] => t, Iterable[t] => t }
+example/MatchType$package.Elem#[X] => typeparam X 
+local0 => case val method t 
```

Comment:
This (and other similar) should be `type t` - they appear like this because the type variables are represented using normal `Bind` trees with TermNames

Reply:
>  the type variables are represented using normal Bind trees with TermNames
Ah, that's why!

Reply:
Just out of curiousity, can I have a pointer that is related to the "the type variables are represented using normal Bind trees with TermNames" part?

Reply:
The first two (`t` of `Array[t]` and `Iterable[t]`) has a TypeSignature `TypeSignature(None,TypeRef(Empty,scala/Nothing#,List()),TypeRef(Empty,scala/Any#,List()))`
We just don't write to `metac.expect` https://github.com/lampepfl/dotty/blob/fa4cb01c1aeb7f6e60b1cf67d29d04ca30ec5ea2/compiler/src/dotty/tools/dotc/semanticdb/PPrint.scala#L131-L137 because they are trivial. Maybe we should write them even if it's trivial to avoid this kind of confusion (even I forgot why nothing has written 😅 )

The last two (symbols for `x *: xs`) have `TypeSignature(None, Empty, Empty)` (that shouldn't happen), because I guess, as you mentioned, those type variables are represented using `Bind` tree with TermNames, and no symbol information is available from the typed tree.

Reply:
here is the Tasty for the `Concat` type in the example, it gives an idea of what the tree looks like:
```scala
TYPEDEF(135) 24 [Concat]
   129:         LAMBDAtpt(131)
   132:           TYPEPARAM(11) 25 [Xs]
   135:             TYPEBOUNDStpt(8)
   137:               SHAREDtype 43
   139:               IDENTtpt 26 [Tuple]
   141:                 TYPEREF 26 [Tuple]
   143:                   SHAREDtype 71
   145:           TYPEPARAM(11) 27 [Ys]
   148:             TYPEBOUNDStpt(7)
   150:               SHAREDtype 43
   152:               IDENTtpt 26 [Tuple]
   154:                 SHAREDtype 141
   157:             COVARIANT
   158:           MATCHtpt(103)
   160:             IDENTtpt 26 [Tuple]
   162:               SHAREDtype 141
   165:             IDENTtpt 25 [Xs]
   167:               TYPEREFdirect 132
   170:             CASEDEF(13)
   172:               IDENTtpt 28 [EmptyTuple]
   174:                 TYPEREF 28 [EmptyTuple]
   176:                   TERMREF 29 [Tuple$package]
   178:                     SHAREDtype 30
   180:               IDENTtpt 27 [Ys]
   182:                 TYPEREFdirect 145
   185:             CASEDEF(76)
   187:               APPLIEDtpt(41)
   189:                 IDENTtpt 30 [*:]
   191:                   TYPEREF 30 [*:]
   193:                     SHAREDtype 71
   195:                 BIND(16) 31 [x]
   198:                   TYPEBOUNDS(8)
   200:                     TYPEREF 14 [Nothing]
   202:                       SHAREDtype 71
   204:                     TYPEREF 15 [Any]
   206:                       SHAREDtype 71
   208:                   IDENT 21 [_]
   210:                     SHAREDtype 198
   213:                 BIND(15) 32 [xs]
   216:                   TYPEBOUNDS(7)
   218:                     SHAREDtype 200
   221:                     TYPEREF 26 [Tuple]
   223:                       SHAREDtype 71
   225:                   IDENT 21 [_]
   227:                     SHAREDtype 216
   230:               APPLIEDtpt(31)
   232:                 IDENTtpt 30 [*:]
   234:                   SHAREDtype 191
   237:                 IDENTtpt 31 [x]
   239:                   TYPEREFdirect 195
   242:                 APPLIEDtpt(19)
   244:                   IDENTtpt 24 [Concat]
   246:                     TYPEREFsymbol 125
   248:                       THIS
   249:                         TYPEREFsymbol 5
   251:                           TERMREFpkg 1 [example]
   253:                   IDENTtpt 32 [xs]
   255:                     TYPEREFdirect 213
   258:                   IDENTtpt 27 [Ys]
   260:                     SHAREDtype 182
```